sharktank/sharktank/layers/kv_cache.py

```diff
@@ -98,7 +98,7 @@ def unflatten_page_table(
             shards = [
                 shard.unflatten(1, self.sub_page_dims) for shard in page_slab.shards
             ]
-            return SplitPrimitiveTensor(ts=shards, shard_dim=4)
+            return SplitPrimitiveTensor(ts=shards, shard_dim=4, devices=tuple(range(self.shard_count)), devices_pinned=True)  # TODO
```

Comment:
Lets avoid changing these sections and use them as essentially tests that the default path works.

One day I could us representing `ReplicatedTensor` differently. Storing the shards in a map instead of a list. Then replicated tensor interactions would lookup the corresponding replicated shard. Not required in the short term.